src/dev/flang/util/SourceFile.java

```diff
@@ -640,10 +640,19 @@ private int[] lines()
   public int codePoint(int pos)
   {
     if (PRECONDITIONS) require
-      (pos >= 0,
-       pos < _bytes.length);
+      (pos >= 0);
```

Comment:
It might be better fail fast here. What I mean is to keep and relax the precondition `pos < _bytes.length` to `pos <= _bytes.length`. This will make it easier to find bugs where `codePoint` is called with pos greater than `bytes.length`.